hathor/sysctl/core/manager.py

```diff
@@ -61,3 +93,141 @@ def set_profiler_stop(self, save_to: str | None) -> None:
         if not save_to:
             save_to = None
         self.manager.stop_profiler(save_to=save_to)
+
+    @signal_handler_safe
+    def set_pudb_set_trace_attach_tty(self, tty: str) -> None:
+        """Stop execution and open pudb in a given tty.
+
+        ATTENTION: This command can be destructive and the full node might not work after running it.
+
+        Open a new terminal. First, you need to get the path of the tty of the
+        terminal you want to debug from. To do that, use the standard unix
+        command `tty`. It will print something like `/dev/pts/3`.
+
+        Then you need to make sure that your terminal doesn't have a shell actively
+        reading and possibly capturing some of the input that should go to pudb.
+        To do that run a placeholder command that does nothing, such as `perl -MPOSIX -e pause`.
+        """
+        fp = open(tty, 'r+b', buffering=0)
+        term_size = os.get_terminal_size(fp.fileno())
+        self._run_pudb_set_trace(tty, fp, term_size=term_size)
+
+    @signal_handler_safe
+    def set_pudb_set_trace_create_tty(self, cols: int, rows: int) -> None:
+        """Stop execution and open pudb for debugging in a newly created tty.
+
+        ATTENTION: This command can be destructive and the full node might not work after running it.
+
+        You must provide the terminal size (cols, rows).
+
+        The newly created tty path will be printed in the logs. After you get it, you can
+        connect to it using `screen <ttyname>`.
```

Comment:
The user does not set `ttyname`. How can he know this name?

Reply:
I improved the documentation. The name ttyname will be logged. So you have to check the logs and get the name there.